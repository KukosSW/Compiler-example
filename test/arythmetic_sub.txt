DECLARE
    a, b, t(10:10000), c
BEGIN
    a := 100 - 0;
    WRITE a;

    a := 10 - 10;
    WRITE a;

    a := 10 - 11;
    WRITE a;

    a := 15 - 10;
    WRITE a;

    b := 20 - a;
    WRITE b;

    c := a - b;
    WRITE c;

    c := b - a;
    WRITE c;

    a := 1000;
    a := a - c;
    WRITE a;

    b := b - b;
    WRITE b;

    WRITE 0;

    t(10) := 100 - 50;
    WRITE t(10);

    t(11) := t(10) - 10;
    WRITE t(11);

    t(12) := t(10) - t(11);
    WRITE t(12);

    a := 13;
    t(a) := t(11) - t(12);
    WRITE t(a);

    b := a;
    a := a + 1;
    t(a) := t(b) - t(12);
    WRITE t(a);

    c := b;
    b := a;
    a := a + 1;
    t(a) := t(c) - t(b);
    WRITE t(a);

    t(c) := t(c) - t(c);
    WRITE t(c);
END