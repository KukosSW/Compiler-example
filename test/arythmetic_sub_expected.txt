100
0
0
5
15
0
10
990
0
0
50
40
10
30
20
10
0
